File: IAIso-v5.0/sdk/python/iaiso/engine.py
import os

class IAIsoEngine:
    def __init__(self, system_id="global-core"):
        self.p = 0.0
        self.back_prop = os.getenv("BACK_PROPAGATION", "true").lower() == "true"

    def update_pressure(self, tokens=0, tools=0):
        # dp/dt = Input - Dissipation
        delta = (tokens * 0.00015) + (tools * 0.08)
        self.p = max(0.0, self.p + delta - 0.02)
        if self.p >= 0.95:
            self.p = 0.0 # Atomic State Purge
            return "RELEASED"
        return "ESCALATED" if self.p >= 0.85 else "OK"

    def magnify(self, content):
        if self.back_prop:
            return f"[MAGNIFIED] {content}" # Recursive Quality Check
        return content

File: IAIso-v5.0/sdk/javascript/iaiso/engine.js
export class IAIsoEngine {
    constructor(systemId = "web-client") {
        this.systemId = systemId;
        this.p = 0.0;
        this.backProp = true; // Default ON per requirements
    }

    updatePressure(tokens = 0, tools = 0) {
        const delta = (tokens * 0.00015) + (tools * 0.08);
        this.p = Math.max(0, this.p + delta - 0.02);

        if (this.p >= 0.95) {
            this.p = 0.0;
            return "RELEASE_TRIGGERED";
        }
        return this.p >= 0.85 ? "ESCALATED" : "OK";
    }

    magnify(output) {
        return this.backProp ? `[IAIso-MAGNIFIED] ${output}` : output;
    }
}
